CONTRIBUTING.md

```diff
@@ -1,18 +1,149 @@
+
+The first one is the work done to get the Pull Request ready for review.
+The other is the review cycle.
+
+#### Draft Phase
+
+If the pull request is simple and short lived, it can be initialized with review request.
+If the pull request is more complex and is being developed over time, it may be benefitial to start it in a `Draft` state.
+This allows other contributors to monitor the progress and volunteer feedback while annotating that the pull request is not yet ready for review.
+
+By the end of this phase, and right before review is requested, it is helpful for the reviewers to have a clean list of commits in the pull request.
+
+In most cases, a single commit per pull request is enough.
+
+Multiple commits should be used when the commit is too large and the scope of changes can be reduced by separating it into multiple commits which are logically self-contained.
+Such commits do not have to pass tests in isolation, and need only to be meaningfully complete for the reviewer to benefit from reading, compared to reviewing all the changes at once.
+
+### Review Phase
+
+Once the pull request is ready, the author can switch from draft to regular pull request.
+
+At this point, the pull request will be triaged during the next triage session and reviewers will be assigned to it.
+
+In this phase, any changes applied to the pull request should result in additive commits to it. This allows reviewers to see what changes have been made in result of their feedback and evaluate them.
+
+### Merging
+
+Every PR requires at least one review to be merged.
+
+If the author has the editing rights to the repository merging should be performed by the author of the pull request. If the author wants to grant another team member rights to merge, they can state so in the PR comment.
+
+If the pull request modifies code in one of the recognized components, one of the component owners should be on the reviewers list for the pull request. For the list of components and their owners, see [CODEOWNERS](CODEOWNERS).
+
+## Review Model
+
+Every project has its own code authoring and review culture which grows organically as the project matures and is a subject to change.
+Below is the description of the model we try to follow, with exceptions when the common sense dictates otherwise.
+
+### Review Types
+
+Review can be of either **architectural** or **technical** value, and often is of both.
+
+The PR author can specify, when requesting review, what kind of review they are asking for, from each reviewer, or even which area they'd like the reviewer to focus on (useful when distributing reviews).
+
+### Reviewer Role
+
+The reviewer is responsible for accepting a pull request only once they feel the current PR is ready to be merged even if their comments were not to be applied.
+
+The *accepted* can be set with pending review comments, if those comments don't affect whether the patch is ready to be merged (for example, they're stylistic suggestions).
```

Comment:
With *accepted*, are you referring to the "Approve" status of a review?  IIUC, you're describing the process of leaving optional comments in a review that is marked "Approve" regardless of the response to those comments.